### PR TITLE
Change output_of to execute commands with shell_tools.run

### PR DESCRIPTION
**Pull request by pavoljuhas**
Stop using `shell_tools.run_cmd()` which is slated for removal.
Get command output with `shell_tools.run()` instead.
Update `output_of()` to receive command-line arguments in
a list or string in order to have the same interface as `run()`.
Remove filtering of the optional None arguments,
leave it up to callers to take care of that.

Partially implements https://github.com/quantumlib/Cirq/issues/4394

### CODE REVIEW

File: dev_tools/shell_tools.py [hunk 1]
```diff
@@ -290,19 +290,19 @@ def output_of(*cmd: Optional[str], **kwargs) -> str:
...
-        to an instance of TeeCapture.
+        str: The standard output of the command with the last newline removed.
 
     Raises:
          subprocess.CalledProcessError: The process returned a non-zero error
-            code and raise_on_fail was set.
+            code and the `check` flag was True (default).
     """
-    result = cast(str, run_cmd(*cmd, log_run_to_stderr=False, out=TeeCapture(), **kwargs).out)
+    cmdpure = [w for w in cmd if w is not None]
```
Comment (verult) [inline dev_tools/shell_tools.py:304]: nit: Should this check be inside `run()`? Do all callers of `run()` need to check this?

File: dev_tools/shell_tools.py [hunk 2]
```diff
@@ -290,19 +290,19 @@ def output_of(*cmd: Optional[str], **kwargs) -> str:
 
     Args:
         *cmd: Components of the command to execute, e.g. ["echo", "dog"].
-        **kwargs: Extra arguments for asyncio.create_subprocess_shell, such as
+            Arguments set to None are disregarded in command execution.
```
Comment (verult) [inline dev_tools/shell_tools.py:293]: What if we change the type of `cmd` to `str` instead of `Optional[str]`?

File: dev_tools/shell_tools.py [hunk 3]
```diff
@@ -290,19 +290,19 @@ def output_of(*cmd: Optional[str], **kwargs) -> str:
...
+        **kwargs: Extra arguments for the shell_tools.run function, such as
             a cwd (current working directory) argument.
 
     Returns:
-        A (captured output, captured error output, return code) triplet. The
-        captured outputs will be None if the out or err parameters were not set
-        to an instance of TeeCapture.
+        str: The standard output of the command with the last newline removed.
```
Comment (verult) [inline dev_tools/shell_tools.py:298]: Are stderr and exit code no longer needed?

File: dev_tools/shell_tools.py (hunk 1 above)
Comment (pavoljuhas) [reply dev_tools/shell_tools.py:304]: I'd like to have a minimum difference between behavior of shell_tools.run and subprocess.run, so I'd rather not do any filtering of the cmd arguments.

For consistency sake it is perhaps better to have output_of() handle arguments in the same way as run() and do the argument filtering in the caller.  I have checked the references to output_of and it seems easily doable.

File: dev_tools/shell_tools.py (hunk 3 above)
Comment (pavoljuhas) [reply dev_tools/shell_tools.py:298]: The original docstring was incorrect, perhaps a result of copy-paste.
The output_of() function returns/returned only the stdout string.

**Comment by pavoljuhas** [comment]
@verult - updated and ready for review.

**Review by verult — APPROVED** [review]
LGTM, thanks Pavol!

**Review by MichaelBroughton — APPROVED** [review]

**Pull request merged**